### PR TITLE
Fix face angles (yaw, pitch, & roll) accuracy

### PR DESCRIPTION
**Pull request by TheSohaibAhmed**
1) Previouly derived angles seemed inaccurate and somewhat counter-intuitive (given their output was in radians). 
This update uses the a person's mesh positions (as were used before), and chooses specific points for more accurate results (which are rounded to the nearest deg). It also adds directionality of the movements (_e.g. pitching head backwards is a negative result, as is rolling head to the left).

2) The webcam.js file has also been updated to showcase the correct output in degrees (reducing potential user confusion)

3) Stylistic changes made to avoid errors arising from prev. commits to the same file (e.g. double quotes were used instead of single quotes)

Author: Sohaib Ahmed <sohaibi.ahmed@icloud.com>

### CODE REVIEW

File: demo/webcam.js [hunk 1]
```diff
@@ -53,12 +53,12 @@ function drawFaces(canvas, data, fps) {
...
-    ctx.fillText(`roll:${person.angle.roll.toFixed(3)} pitch:${person.angle.pitch.toFixed(3)} yaw:${person.angle.yaw.toFixed(3)}`, person.detection.box.x, person.detection.box.y - 5);
+    ctx.fillText(`roll:${person.angle.roll}° pitch:${person.angle.pitch}° yaw:${person.angle.yaw}°`, person.detection.box.x, person.detection.box.y - 5);
     ctx.fillStyle = 'lightblue';
     ctx.fillText(`gender: ${Math.round(100 * person.genderProbability)}% ${person.gender}`, person.detection.box.x, person.detection.box.y - 60);
     ctx.fillText(`expression: ${Math.round(100 * expression[0][1])}% ${expression[0][0]}`, person.detection.box.x, person.detection.box.y - 42);
     ctx.fillText(`age: ${Math.round(person.age)} years`, person.detection.box.x, person.detection.box.y - 24);
-    ctx.fillText(`roll:${person.angle.roll.toFixed(3)} pitch:${person.angle.pitch.toFixed(3)} yaw:${person.angle.yaw.toFixed(3)}`, person.detection.box.x, person.detection.box.y - 6);
+    ctx.fillText(`roll:${person.angle.roll}° pitch:${person.angle.pitch}° yaw:${person.angle.yaw}°`, person.detection.box.x, person.detection.box.y - 6);
```
Comment (vladmandic) [inline demo/webcam.js:61]: what's the point of the change if angles are already result of `Math.floor()` function so they do not have decimal values?

File: src/factories/WithFaceLandmarks.ts [hunk 2]
```diff
@@ -5,64 +5,130 @@ import { isWithFaceDetection, WithFaceDetection } from './WithFaceDetection';
...
-    && obj['alignedRect'] instanceof FaceDetection;
+    && obj['alignedRect'] instanceof FaceDetection
+  );
 }
 
 function calculateFaceAngle(mesh) {
-  // returns the angle in the plane (in radians) between the positive x-axis and the ray from (0,0) to the point (x,y)
-  const radians = (a1, a2, b1, b2) => (Math.atan2(b2 - a2, b1 - a1) % Math.PI);
-  // convert radians to degrees
+  /*
```
Comment (vladmandic) [inline src/factories/WithFaceLandmarks.ts:35]: I do not include authoring notes in the code itself. Authors for a specific part of the code are visible via git blame, history and included as contributors on github itself.

File: src/factories/WithFaceLandmarks.ts [hunk 3]
```diff
@@ -5,64 +5,130 @@ import { isWithFaceDetection, WithFaceDetection } from './WithFaceDetection';
...
+
+  angle.roll = calcRoll(pt[27], pt[66]);
+  angle.pitch = calcPitch(pt[14], pt[30], pt[2]);
+  angle.yaw = calcYaw(pt[14], pt[33], pt[2]);
 
   return angle;
 }
 
-export function extendWithFaceLandmarks<TSource extends WithFaceDetection<{}>, TFaceLandmarks extends FaceLandmarks = FaceLandmarks68 >(sourceObj: TSource, unshiftedLandmarks: TFaceLandmarks): WithFaceLandmarks<TSource, TFaceLandmarks> {
+export function extendWithFaceLandmarks<
```
Comment (vladmandic) [inline src/factories/WithFaceLandmarks.ts:113]: word-wrapping is good for extra long lines, but this is a bit extra

File: src/factories/WithFaceLandmarks.ts [hunk 4]
```diff
@@ -5,64 +5,130 @@ import { isWithFaceDetection, WithFaceDetection } from './WithFaceDetection';
...
+  TSource extends WithFaceDetection<{}>,
+  TFaceLandmarks extends FaceLandmarks = FaceLandmarks68
+>(
+  sourceObj: TSource,
+  unshiftedLandmarks: TFaceLandmarks,
+): WithFaceLandmarks<TSource, TFaceLandmarks> {
   const { box: shift } = sourceObj.detection;
-  const landmarks = unshiftedLandmarks.shiftBy<TFaceLandmarks>(shift.x, shift.y);
+  const landmarks = unshiftedLandmarks.shiftBy<TFaceLandmarks>(
```
Comment (vladmandic) [inline src/factories/WithFaceLandmarks.ts:121]: unnecessary change.

**Review by vladmandic — CHANGES_REQUESTED** [review]

File: demo/webcam.js (hunk 1 above)
Comment (TheSohaibAhmed) [reply demo/webcam.js:61]: Because since the angles are rounded off now, we don't need / want three trailing zeroes behind it (_i.e._ 12 deg prev. looked like 12.000, which didn't help + cluttered the screen)

File: src/factories/WithFaceLandmarks.ts (hunk 3 above)
Comment (TheSohaibAhmed) [reply src/factories/WithFaceLandmarks.ts:113]: ^That was one of the lint recommendation: the file had a bunch of errors, so I tried to fix those including this one

File: src/factories/WithFaceLandmarks.ts (hunk 4 above)
Comment (TheSohaibAhmed) [reply src/factories/WithFaceLandmarks.ts:121]: Same response as above: change made to remove lint error(s)

**Comment by vladmandic** [comment]
looks good, i'll probably merge tomorrow.
